modules/ballerina-core/src/main/java/org/ballerinalang/model/values/BArray.java

```diff
@@ -172,9 +172,10 @@ private void grow(int capacityRequired, int bucketIndex) {
         }
 
         if (bucketIndex >= arrayBucket.length) {
+            int newLength = arrayBucket.length + (DEFAULT_ARRAY_BUCKET_SIZE > (bucketIndex - arrayBucket.length) ?
+                    DEFAULT_ARRAY_BUCKET_SIZE : (bucketIndex - arrayBucket.length)) + 1;
```

Comment:
Is the change given below ok for this?

`int newLength = (DEFAULT_ARRAY_BUCKET_SIZE > (bucketIndex - arrayBucket.length)) ?
                    (arrayBucket.length + DEFAULT_ARRAY_BUCKET_SIZE) : (bucketIndex + DEFAULT_ARRAY_BUCKET_SIZE);`